docs/csharp/language-reference/builtin-types/nullable-reference-types.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Nullable reference types - C# reference"
+description: Learn about C# nullable reference types and how to use them
+ms.date: 04/06/2020
+---
+# Nullable reference types (C# reference)
+
+> [!NOTE]
+> This article covers nullable reference types. You can also declare [nullable value types](nullable-value-types.md).
+
+Nullable reference types are available beginning with C# 8.0, in code that has opted in to a *nullable aware context*. Nullable reference types, the null static analysis warnings, and the [null-forgiving operator](../operators/null-forgiving.md) are optional language features. All are turned off by default. A *nullable context* is controlled at the project level using build settings, or in code using pragmas.
+
+ In a nullable aware context:
+
+- A variable of a reference type `T` must be initialized with non-null, and may never be assigned a value that may be `null`.
+- A variable of a reference type `T?` may be initialized with `null` or assigned `null`, but is required to be checked against `null` before de-referencing.
+- A variable `m` of type `T?` is considered to be non-null when you apply the null-forgiving operator, as in `m!`.
+
+The distinctions between a non-nullable reference type `T` and a nullable reference type `T?` are performed by the compiler's interpretation of the preceding rules. A variable of type `T` and a variable of type `T?` are represented by the same .NET type. The following example declares a non-nullable string, a nullable string, and uses the null-forgiving operator to assign a value to a non-nullable string:
+
+:::code language="csharp" source="snippets/NullableReferenceTypes.cs" id="SnippetCoreSyntax":::
+
+The variables `notNull` and `nullable` are both represented by the <xref:System.String> type. Because the non-nullable and nullable types are both stored as the same type, there are several locations where using a nullable reference type isn't allowed. In general, a nullable reference type can't be used as a base class or implemented interface. A nullable reference type can't be used in any object creation or type testing expression. A nullable reference type can't be the type of a member access expression. The following examples show these constructs:
```

Comment:
>The following examples show these constructs:

I don't see the implemented interface example.

I think this might be easier to follow if it were a bulleted list with the description of each example preceding the code for that scenario.